File: Miscelaneous/Emprestimo.java
import java.time.LocalDate;

public class Emprestimo {
    private final Livro livro;
    private final String aluno;
    private final LocalDate dataRetirada;

    public Emprestimo(Livro livro, String aluno) {
        if (!livro.emprestar()) throw new IllegalStateException("Livro indisponivel para emprestimo.");
        this.livro = livro;
        this.aluno = aluno;
        this.dataRetirada = LocalDate.now();
    }

    public void devolver() { livro.devolver(); }

    public void exibirResumo() {
        System.out.println("Emprestimo: " + livro.getTitulo() + " (" + livro.getAutor() + ") para " + aluno + " em " + dataRetirada);
    }
}

File: Miscelaneous/emprestimo.py
from datetime import date


class Emprestimo:
    def __init__(self, livro, aluno):
        if not livro.emprestar():
            raise ValueError("Livro indisponivel para emprestimo.")
        self.livro = livro
        self.aluno = aluno
        self.data_retirada = date.today()

    def devolver(self):
        self.livro.devolver()

    def exibir_resumo(self):
        print(f"Emprestimo: {self.livro.titulo} ({self.livro.autor}) para {self.aluno} em {self.data_retirada}")